src/main/java/com/amazon/ion/impl/IonCursorBinary.java

```diff
@@ -132,6 +132,11 @@ private static class RefillableState {
 
     }
 
+    /**
+     * Dummy state that indicates the cursor has been terminated and that additional API calls will have no effect.
+     */
+    private static final RefillableState TERMINATED_STATE = new RefillableState(null, -1, -1);
```

Comment:
It seems sensible, even if it's not essential, for the `TERMINATED_STATE` to have a pre-populated state of `TERMINATED`.
```suggestion
    private static final RefillableState TERMINATED_STATE = new RefillableState(null, -1, -1);
    static {
        TERMINATED_STATE.state = State.TERMINATED;
    }
```

Reply:
Done.